profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/LinuxStackFramesCollector.cpp

```diff
@@ -82,7 +95,15 @@ StackSnapshotResultBuffer* LinuxStackFramesCollector::CollectStackSampleImplemen
         }
         else
         {
-            _stackWalkInProgressWaiter.wait(stackWalkInProgressLock);
+            do
```

Comment:
I would add a small comment here to explain why we needed this code; something like:
   // When the application ends and the CLR shuts down, it might happen that 
   // the currently walked thread gets terminated without notice for us.
   // This loop ensures that the code does not stay stuck waiting for a lock that will
   // never be released. It will exit if the stack walked thread does not run anymore or 
   // the stack walk finishes as expected.
Feel free to adapt  :^)